cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/ReducerWrapper.java

```diff
@@ -69,6 +76,15 @@ public void run(Context context) throws IOException, InterruptedException {
     WrappedReducer.Context flushingContext = createAutoFlushingContext(context, basicMapReduceContext);
     basicMapReduceContext.setHadoopContext(flushingContext);
 
+    int metricsReportInterval = context.getConfiguration().getInt(TaskMetricsWriter.PROPERTY_NAME, 60);
+    final ReduceTaskMetricsWriter reduceTaskMetricsWriter = new ReduceTaskMetricsWriter(basicMapReduceContext, context);
+    metricsEmitterService.scheduleAtFixedRate(new Runnable() {
```

Comment:
It's better not to start a new thread as I am not sure if the task context is thread safe or not. It's better do it in the `WrappedMapper.Context` and `WrappedReducer.Context` when data are read/write.